### PR TITLE
[docs] 1/n rm `airflow-dagster` reference in prep. for migration guides

### PR DESCRIPTION
**Pull request by cmpadden**
## Summary & Motivation

- Removes `/integrations/airflow/reference` as we do not want to promote the use of the `dagster-airflow` integration
- Prepares us for "Pick your own journey" Airflow migration guides

### CODE REVIEW

**Comment by cmpadden** [comment]
* **#22612** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/22612?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @cmpadden and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/content/integrations/airflow.mdx [hunk 1]
```diff
@@ -1,15 +1,11 @@
...
 ---
 
 # Airflow & Dagster
 
-<CodeReferenceLink filePath="examples/with_airflow" />
-
-The [`dagster-airflow`](/\_apidocs/libraries/dagster-airflow) library provides interoperability between Dagster and Airflow. This integration is designed to help support users who have existing Airflow usage and are interested in exploring Dagster.
-
-Using `dagster-airflow`, you can incorporate all or some of your existing Airflow Directed Acyclic Graphs (DAGs) or orchestrate Dagster jobs from Airflow. Refer to the [Airflow reference](/integrations/airflow/reference) for examples or the following sections for info on migrating from Airflow.
+Migrating from Airflow to Dagster, or integrating Dagster into your existing workflow orchestration stack can be accomplished in many ways. The <a href="/guides/migrations/migrating-airflow-to-dagster">Pick your own journey</a> guide provides a variety suggestions in how one can migrate their Airflow pipelines to Dagster, or build a platform where both tools co-exist.
```
Comment (maximearmstrong) [inline docs/content/integrations/airflow.mdx:8]: nit: I'd add a comma to ease readability

```suggestion
Migrating from Airflow to Dagster, or integrating Dagster into your existing workflow orchestration stack, can be accomplished in many ways. The <a href="/guides/migrations/migrating-airflow-to-dagster">Pick your own journey</a> guide provides a variety suggestions in how one can migrate their Airflow pipelines to Dagster, or build a platform where both tools co-exist.
```

**Review by maximearmstrong — APPROVED** [review]
That's great. I added one nit.

File: docs/content/integrations/airflow.mdx (hunk 1 above)
Comment (erinkcochran87) [reply docs/content/integrations/airflow.mdx:8]: +1

Comment (erinkcochran87) [inline docs/content/integrations/airflow.mdx:8]:
```suggestion
Migrating from Airflow to Dagster, or integrating Dagster into your existing workflow orchestration stack can be accomplished in many ways. The [Pick your own journey](/guides/migrations/migrating-airflow-to-dagster) guide provides a variety suggestions in how one can migrate their Airflow pipelines to Dagster, or build a platform where both tools co-exist.
```

No need to use HTML link formatting here - Markdown works just fine.

**Review by erinkcochran87 — APPROVED** [review]
ty!

We should add a redirect for the Airflow reference, otherwise we may end up with broken links somewhere. It could redirect to `/integrations/airflow` to keep things simple.

**Review by yuhan — APPROVED** [review]
should we set up redirect from "/integrations/airflow/reference" to "/integrations/airflow"?

otherwise, it looks great.

**Review by erinkcochran87 — APPROVED** [review]

**Pull request merged**